pure_pursuit_wrapper/src/pure_pursuit_wrapper.cpp

```diff
@@ -14,42 +14,185 @@
  * the License.
  */
 
-#include "pure_pursuit_wrapper/pure_pursuit_wrapper.hpp"
-#include <trajectory_utils/conversions/conversions.h>
-#include <carma_wm/Geometry.h>
+#include <pure_pursuit_wrapper/pure_pursuit_wrapper.hpp>
+#include <trajectory_utils/conversions/conversions.hpp>
+#include <carma_wm_ros2/Geometry.hpp>
 #include <algorithm>
 
 
 namespace pure_pursuit_wrapper
 {
-PurePursuitWrapper::PurePursuitWrapper(PurePursuitWrapperConfig config, WaypointPub waypoint_pub, PluginDiscoveryPub plugin_discovery_pub)
-  : config_(config), waypoint_pub_(waypoint_pub), plugin_discovery_pub_(plugin_discovery_pub)
+namespace std_ph = std::placeholders;
+
+PurePursuitWrapperNode::PurePursuitWrapperNode(const rclcpp::NodeOptions& options)
+  : carma_guidance_plugins::ControlPlugin(options)
+{
+  config_ = PurePursuitWrapperConfig();
+  config_.vehicle_response_lag = declare_parameter<double>("vehicle_response_lag", config_.vehicle_response_lag);
+  config_.minimum_lookahead_distance = declare_parameter<double>("minimum_lookahead_distance", config_.minimum_lookahead_distance);
+  config_.maximum_lookahead_distance = declare_parameter<double>("maximum_lookahead_distance", config_.maximum_lookahead_distance);
+  config_.speed_to_lookahead_ratio = declare_parameter<double>("speed_to_lookahead_ratio", config_.speed_to_lookahead_ratio);
+  config_.is_interpolate_lookahead_point = declare_parameter<bool>("is_interpolate_lookahead_point", config_.is_interpolate_lookahead_point);
+  config_.is_delay_compensation = declare_parameter<bool>("is_delay_compensation", config_.is_delay_compensation);
+  config_.emergency_stop_distance = declare_parameter<double>("emergency_stop_distance", config_.emergency_stop_distance);
+  config_.speed_thres_traveling_direction = declare_parameter<double>("speed_thres_traveling_direction", config_.speed_thres_traveling_direction);
+  config_.dist_front_rear_wheels = declare_parameter<double>("dist_front_rear_wheels", config_.dist_front_rear_wheels);
+
+  // integrator part
+  config_.dt = declare_parameter<double>("dt", config_.dt);
+  config_.integrator_max_pp = declare_parameter<double>("integrator_max_pp", config_.integrator_max_pp);
+  config_.integrator_min_pp = declare_parameter<double>("integrator_min_pp", config_.integrator_min_pp);
+  config_.Ki_pp = declare_parameter<double>("Ki_pp", config_.Ki_pp);
+  config_.integral = declare_parameter<double>("integral", config_.integral);
```

Comment:
Is this really a parameter? In the autoware PR it looked like the accumulator for the sum of errors for use in the PID

Reply:
correct I removed it from here